MFA (#740)
neo4j/docs-aura: modules/ROOT/pages/security/mfa.adoc

= Multi-Factor Authentication 

Multi-Factor Authentication (MFA) adds an extra layer of security to an Aura account log-in by requiring a verification code in addition to username and password. 
The verification code is generated using any authenticator app, such as Google Authenticator.

There are two scenarios for MFA:

* Individual users can enable MFA for their own accounts via account settings.
* Organization administrators can enable MFA for an entire organization, then it's mandatory for all members to go through the MFA setup.

[IMPORTANT]
====
Setting up MFA requires logging in using email and password, not using SSO or Google Sign-In.
====

> == Enable MFA as a user via account settings 

Reviewer: ```suggestion
== Enable individual MFA
```

. Log in to the Aura Console using email/password.

> . Select *Account > Settings > Preferences > Security*.

Reviewer: ```suggestion
. Go to *Account > Settings > Preferences > Security*.
```

. Enable *Multi-Factor Authentication (MFA)*.
. Follow the MFA set up steps.

==  Enable MFA for an organization 

`Organization Owners` or `Organization Admins` can require all members of an organization to set up MFA from organization security settings.
Then each organization member will be prompted to complete the setup, with an authenticator app of their choice.
 

> To require MFA organization wide:

Reviewer: ```suggestion
To set up organization-wide MFA:
```

. Go to *Organization Settings > Security & Networking > App MFA (Multi-Factor Authentication)*.
. Enable *Multi-Factor Authentication (MFA)*.

After MFA is enabled at the organization level, all members will see the following message in the console:

[quote]
"_Your organization has required Multi Factor Authentication (MFA). 
Clicking enable MFA below will log you out and you will then need to login using your email and password to configure MFA._"

> Selecting *Enable*, logs the user out automatically, and then it's required to continue with MFA setup steps.

Reviewer: ```suggestion
Selecting *Enable*, logs the user out automatically, and then they are required to continue with MFA setup steps.
These are the same as for an individual user setting up MFA for their own account, as described above.
```

> == MFA set up steps
> 
> . Log in using email and password.
> . Enter the one-time code sent to your email account.
> . Use an authenticator app to scan the QR code in the setup modal.
> . Enter the verification code from the authenticator app.

Reviewer: ```suggestion
```

= Multi-Factor Authentication 

Multi-Factor Authentication (MFA) adds an extra layer of security to your Neo4j Aura account by requiring a verification code in addition to your password. 
To generate the verification code use an authenticator app, such as Google Authenticator.

> There's two scenarios for MFA:

Reviewer: ```suggestion
There are two scenarios for MFA:
```

* Scenario 1: Individual users can enable MFA for their own accounts.
* Scenario 2: Organization administrators can enable MFA for an entire organization, then it's mandatory for all members to go through the MFA setup.

== Scenario 1. Enable MFA as a user via account settings 

[IMPORTANT]
====
To enable MFA, you must log in using email and password, not Google login or SSO.
====

. Log in to the Aura Console using your email/password.
. Go to *Account > Settings > Preferences > Security*.
. In the *Multi-Factor Authentication (MFA)* section select *Enable*.
. You will be logged out automatically.
. Log back in with your email and password.
. A setup modal appears with a QR code.
. Scan the QR code.
. Enter the verification code from the authenticator app.
. Save your recovery code somewhere secure.

==  Scenario 2. Enable MFA for an organization 

> 

Reviewer: ```suggestion
Enabling MFA for all members of an organization consists of two parts:

```

* `Organization Owners` or `Organization Admins` can enable MFA that applies to all members of the organization.

> * Members must log in using email and password. MFA setup is not supported for accounts authenticated through SSO or Google OAuth.

Reviewer: ```suggestion
* * Organization members complete the setup on their end with an authenticator app of their choice.
```

> === To require MFA for all organization members:

Reviewer: ```suggestion
For organization owner or organization admin to enable MFA for their organization:
```

. Go to *Account > Settings > Preferences > Security*.

> . In the *Multi-Factor Authentication (MFA)* section select *Enable*.

Reviewer: ```suggestion
. Enable *Multi-Factor Authentication (MFA)*.
```

> ==== Member steps to complete MFA setup:

Reviewer: ```suggestion
```

> . Organization members will see the message, "_Your organization has required Multi Factor Authentication (MFA). Clicking enable MFA below will log you out and you will then need to login using your email and password to configure MFA._"

Reviewer: ```suggestion
Once MFA has been enabled on the organization level, as an *organization member*, you need to complete the setup on your end.
Next time they log in, they will see the message, "_Your organization has required Multi Factor Authentication (MFA). 
Clicking enable MFA below will log you out and you will then need to login using your email and password to configure MFA._"
```

> . They click *Enable*.
> . Are logged out automatically.
> . Log back in using their email and password.
> . A setup modal appears with a QR code.
> . Using their authenticator app, they scan the QR code.
> . Enter the verification code from the authenticator app.
> . Save the recovery code somewhere secure.

Reviewer: ```suggestion

. Once you click *Enable*, you are logged out automatically.
. Log back in using your email and password, SSO is not supported for MFA setup.
. Use your authenticator app to scan the QR code in the setup modal.
. Enter the verification code from the authenticator app.
. Save the recovery code somewhere secure.
```
